bugwarrior/config.py

```diff
@@ -184,9 +185,27 @@ def get_taskrc_path(conf, main_section):
     )
 
 
-def get_data_path():
-    return os.path.expanduser(os.getenv('TASKDATA', '~/.task'))
+@lru_cache()
+def get_data_path(config, main_section):
+    taskrc = get_taskrc_path(config, main_section)
 
+    # We cannot use the taskw module here because it doesn't really support
+    # the `_` subcommands properly (`rc:` can't be used for them).
+    line_prefix = 'data.location='
+    tw_show = subprocess.Popen(
+        ('task', '_show'), stdout=subprocess.PIPE, env={'TASKRC': taskrc})
+    data_location = subprocess.check_output(
+        ('grep', line_prefix), stdin=tw_show.stdout)
+    tw_show.wait()
+    data_path = data_location[len(line_prefix):].rstrip()
+
+    if not data_path:
+        data_path = os.getenv('TASKDATA', '~/.task')
+        log.warning(
+            'Unable to determine the data location. Falling back to %s.' %
+            data_path)
```

Comment:
If `task` isn't in the path, how are any other `taskw.TaskWarriorShellout` commands going to work? Bugwarrior doesn't offer any other way to find the binary.

The `_show` output is part of task's shell completion support; I doubt it will change that much since it's likely to stay simple.